pkg/controller/replication_controller_test.go

```diff
@@ -375,3 +418,49 @@ func TestWatchControllers(t *testing.T) {
 		t.Errorf("Expected 1 call but got 0")
 	}
 }
+
+// Test that values seen by the reflector are acted upon by the controller
+func TestReplicationControllerReflectorListPods(t *testing.T) {
+	body, _ := latest.Codec.Encode(newPodList(2, testLabels))
+	fakeHandler := util.FakeHandler{
+		StatusCode:   200,
+		ResponseBody: string(body),
+	}
+	testServer := httptest.NewServer(&fakeHandler)
+	defer testServer.Close()
+	client := client.NewOrDie(&client.Config{Host: testServer.URL, Version: testapi.Version()})
+
+	fakePodControl := FakePodControl{}
+
+	manager := NewReplicationManager(client)
+	manager.podControl = &fakePodControl
+
+	numReplicas := 1
+	controllerSpec := newReplicationController(numReplicas)
+	if err := waitForPods(manager.podLister, 2, reflectorTimeout); err != nil {
+		t.Errorf("Reflector took longer than 100ms to populate store %v", err)
```

Comment:
If you use reflectorTimeout in waitForPods, then I think you should also use it in the error message (or do not write timeout at all)